### PR TITLE
[ADF-4566]Refactor cardViewComponentPage and add a new test for priority field …

### PR DESCRIPTION
**Pull request by cristinaj**
…validation.

**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/ADF-4566

### CODE REVIEW

File: lib/testing/src/lib/core/pages/card-view-items/card-view-text-item.page.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    enterTextField(text) {
+        const input = this.item.element(this.editInputLocator);
+        BrowserVisibility.waitUntilElementIsVisible(input);
+        BrowserActions.clearSendKeys(input, text);
+        return this;
+    }
+
+    getFieldErrorMessage() {
+        const errorMessage = this.item.element(this.errorMessageLocator);
```
Comment (eromano) [inline lib/testing/src/lib/core/pages/card-view-items/card-view-text-item.page.ts]: BrowserActions.getText

File: lib/testing/src/lib/core/pages/card-view-items/card-view-text-item.page.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    getTextFieldText() {
+        BrowserVisibility.waitUntilElementIsVisible(this.item);
+        return BrowserActions.getText(this.item.element(this.textValueLocator));
+    }
+
+    enterTextField(text) {
+        const input = this.item.element(this.editInputLocator);
+        BrowserVisibility.waitUntilElementIsVisible(input);
+        BrowserActions.clearSendKeys(input, text);
```
Comment (eromano) [inline lib/testing/src/lib/core/pages/card-view-items/card-view-text-item.page.ts]: BrowserActions. waitUntilElementIsVisible is already in  clearSendKeys

File: lib/testing/src/lib/core/pages/card-view-items/card-view-text-item.page.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        return this;
+    }
+
+    clickOnTextSaveIcon() {
+        BrowserActions.click(this.item.element(this.saveIconLocator));
+        return this;
+    }
+
+    getTextFieldText() {
+        BrowserVisibility.waitUntilElementIsVisible(this.item);
```
Comment (eromano) [inline lib/testing/src/lib/core/pages/card-view-items/card-view-text-item.page.ts]: waitUntilElementIsVisible isz already in  getText

File: e2e/pages/adf/demo-shell/cardViewDemoPage.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    getOutputText(index) {
+        return BrowserActions.getText(this.consoleLog.all(by.css('p')).get(index));
+    }
+
+    disableEdit() {
+        BrowserVisibility.waitUntilElementIsVisible(this.editableSwitch);
+
+        this.editableSwitch.getAttribute('class').then((check) => {
```
Comment (eromano) [inline e2e/pages/adf/demo-shell/cardViewDemoPage.ts]: instead to repead this pattern can we use the one in FormControllersPage maybe moving FormControllersPage under the material fodler in testing

File: lib/testing/src/lib/core/pages/card-view-items/card-view-keyvaluepairs-item.page.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    checkEditIconIsNotDisplayed() {
+        BrowserVisibility.waitUntilElementIsNotVisible(this.item.element(this.editIcon));
+    }
+
+    setNameOnRowByNumber(name, rowNumber) {
+        const nameInputField = this.getRowByNumber(rowNumber).element(this.nameInputLocator);
+        BrowserVisibility.waitUntilElementIsVisible(nameInputField);
+        nameInputField.sendKeys(name);
```
Comment (gmandakini) [inline lib/testing/src/lib/core/pages/card-view-items/card-view-keyvaluepairs-item.page.ts]: can we use clearSendKeys() of BrowserActions here?

File: lib/testing/src/lib/core/pages/card-view-items/card-view-keyvaluepairs-item.page.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    rowLocator = by.className('adf-card-view__key-value-pairs__row ng-star-inserted');
+    editIcon = by.css('mat-icon[data-automation-id*="card-textitem-edit-icon"]');
+    item;
+
+    constructor(item) {
+        this.item = element(by.xpath(`//div[@data-automation-id="card-key-value-pairs-label-${item}"]/ancestor::adf-card-view-boolitem`));
+    }
+
+    getLabelValue() {
+        return this.item.element(by.css('div[data-automation-id*="label"]')).getText();
```
Comment (gmandakini) [inline lib/testing/src/lib/core/pages/card-view-items/card-view-keyvaluepairs-item.page.ts]: Can we use getText() of BrowserActions here?

File: lib/testing/src/lib/core/pages/card-view-items/card-view-bool-item.page.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+    checkbox = by.css(`mat-checkbox[data-automation-id='card-boolean-boolean']`);
+    item;
+
+    constructor(item) {
+        this.item = element(by.xpath(`//div[@data-automation-id="card-boolean-label-${item}"]/ancestor::adf-card-view-boolitem`));
+    }
+
+    getLabelValue() {
+        return this.item.element(by.css('div[data-automation-id*="label"]')).getText();
```
Comment (gmandakini) [inline lib/testing/src/lib/core/pages/card-view-items/card-view-bool-item.page.ts]: getText() of BrowserActions here?

File: lib/testing/src/lib/core/pages/card-view-items/card-view-keyvaluepairs-item.page.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        const nameInputField = this.getRowByNumber(rowNumber).element(this.nameInputLocator);
+        BrowserVisibility.waitUntilElementIsVisible(nameInputField);
+        nameInputField.sendKeys(name);
+        return this;
+    }
+
+    setValueOnRowByNumber(value, rowNumber) {
+        const valueInputField = this.getRowByNumber(rowNumber).element(this.valueInputLocator);
+        BrowserVisibility.waitUntilElementIsVisible(valueInputField);
+        valueInputField.sendKeys(value);
```
Comment (gmandakini) [inline lib/testing/src/lib/core/pages/card-view-items/card-view-keyvaluepairs-item.page.ts]: Can we reuse BrowserActions.clearSendKeys here.

File: lib/testing/src/lib/core/pages/card-view-items/card-view-text-item.page.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    errorMessageLocator = by.css(`mat-error[data-automation-id*='card-textitem-error'] li`);
+    editIcon = by.css('mat-icon[data-automation-id*="card-textitem-edit-icon"]');
+    item;
+
+    constructor(item) {
+        this.item = element(by.xpath(`//div[@data-automation-id="card-textitem-label-${item}"]/ancestor::adf-card-view-textitem`));
+    }
+
+    getLabelValue() {
+        return this.item.element(by.css('div[data-automation-id*="label"]')).getText();
```
Comment (gmandakini) [inline lib/testing/src/lib/core/pages/card-view-items/card-view-text-item.page.ts]: The getText methods be used from BrowserActions by passing the elementFinder in most of the places.

**Comment by mauriziovitale** [comment]
Reopen it in the case is needed

**Pull request closed without merging**